test/postprocessing/watermarkProcessor.test.js

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint mocha/no-mocha-arrows: "off" */
+
+'use strict';
+
+const assert = require('assert');
+const fs = require("fs-extra");
+const path = require("path");
+
+const { WatermarkProcessor } = require('../../lib/postprocessing/watermarkProcessor');
+const { getDimensions } = require('../../lib/postprocessing/assetProperties');
+
+describe("watermarkProcessor.js", () => {
+    beforeEach(async function () {
+        process.env.WORKER_BASE_DIRECTORY = 'build/work';
+        await fs.mkdirs('build/work', { recursive: true });
+    });
+
+    afterEach(() => {
+        delete process.env.WORKER_BASE_DIRECTORY;
+        delete process.env.ASSET_COMPUTE_NO_METADATA_IN_IMG;
+    });
+
+    describe('resizing', () => {
+        it('valid - square widthPercent at 100% ', async () => {
```

Comment:
Can we have all those tests as worker unit tests instead? Makes them easier to manage I guess.

Reply:
FYI: If you try SDK tests on Windows, most tests in the worker fail: Windows doesn't resolve symlinks, it considers them being corrupt files. It's therefore a bit inconvenient if there would ever be users having to debug/modify the SDK from a Windows machine.
(Also tests that actually verify some validation without needing to generate an image could stay as unit tests here, couldn't they? @alexkli).

Reply:
This hasn't changed: there are no plans to run SDK run on Windows, it's environment are only Linux containers.

Reply:
Ack - then please ignore my previous FYI (Context: I was thinking about people contributing to SDK using Windows).